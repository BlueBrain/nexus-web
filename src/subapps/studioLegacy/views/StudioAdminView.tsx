import * as React from 'react';
import { useParams } from 'react-router';
import { Link } from 'react-router-dom';

import StudioListContainer from '../containers/StudioListContainer';

const StudioAdminView: React.FunctionComponent<{}> = () => {
  // @ts-ignore
  const { orgLabel, projectLabel } = useParams();
  return (
    <div className="view-container">
      <div className="global-studio-list">
        <div
          className="project-banner no-bg"
          style={{
            marginBottom: 20,
          }}
        >
          <div className="label">
            <h1 className="name">
              <span>
                <Link role="link" to={`/orgs/${orgLabel}`}>
                  {orgLabel}
                </Link>
                {' | '}
                <Link role="link" to={`/orgs/${orgLabel}/${projectLabel}`}>
                  {projectLabel}
                </Link>
              </span>
            </h1>
          </div>
        </div>
        {orgLabel && projectLabel && (
          <StudioListContainer
            orgLabel={orgLabel}
            projectLabel={projectLabel}
          />
        )}
      </div>
    </div>
  );
};

export default StudioAdminView;
