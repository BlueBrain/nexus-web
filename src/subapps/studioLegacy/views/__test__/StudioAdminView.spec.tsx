import { NexusProvider } from '@bbp/react-nexus';
import { createNexusClient } from '@bbp/nexus-sdk';
import StudioAdminView from '../StudioAdminView';
import { createBrowserHistory } from 'history';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import configureStore from '../../../../shared/store';
import fetch from 'node-fetch';
import {
  render,
  fireEvent,
  waitFor,
  screen,
  server,
} from '../../../../utils/testUtil';
import { rest } from 'msw';
import '@testing-library/jest-dom';
import { act } from 'react-dom/test-utils';

describe('StudioListContainer', () => {
  const history = createBrowserHistory({ basename: '/' });

  // establish API mocking before all tests
  beforeAll(() => {
    server.listen();
  });
  // reset any request handlers that are declared as a part of our tests
  // (i.e. for testing one-time error scenarios)
  afterEach(() => server.resetHandlers());
  // clean up once the tests are done
  afterAll(() => server.close());

  const nexus = createNexusClient({
    fetch,
    uri: 'https://localhost:3000',
  });
  const store = configureStore(history, { nexus }, {});
  it('renders studios in a list', async () => {
    await act(async () => {
      await render(
        <Provider store={store}>
          <ConnectedRouter history={history}>
            <NexusProvider nexusClient={nexus}>
              <StudioAdminView></StudioAdminView>
            </NexusProvider>
          </ConnectedRouter>
        </Provider>
      );
    });
    expect(true).toBe(true);
  });
});
