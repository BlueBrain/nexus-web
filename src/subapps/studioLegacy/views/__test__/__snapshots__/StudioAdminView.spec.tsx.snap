// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudioAdminView renders studios in a list 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="view-container"
      >
        <div
          class="global-studio-list"
        >
          <div
            class="project-banner no-bg"
            style="margin-bottom: 20px;"
          >
            <div
              class="label"
            >
              <h1
                class="name"
              >
                <span>
                  <a
                    href="/admin/org"
                    role="link"
                  >
                    org
                  </a>
                   | 
                  <a
                    href="/admin/org/project"
                    role="link"
                  >
                    project
                  </a>
                </span>
              </h1>
            </div>
          </div>
          <div
            class="studio-list"
            id="studio-list"
          >
            <div
              class="studio-modal"
            >
              <button
                class="ant-btn ant-btn-primary ant-btn-block"
                type="button"
              >
                <span
                  aria-label="plus-square"
                  class="anticon anticon-plus-square"
                  role="img"
                  style="font-size: 16px; color: white;"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="plus-square"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M328 544h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8z"
                    />
                    <path
                      d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
                    />
                  </svg>
                </span>
                <span>
                  Create Studio
                </span>
              </button>
            </div>
            <br />
            <div
              class="infinite-search"
              data-testid="infinite-search"
            >
              <div
                class="search"
              >
                <span
                  class="ant-input-group-wrapper ant-input-search"
                >
                  <span
                    class="ant-input-wrapper ant-input-group"
                  >
                    <span
                      class="ant-input-affix-wrapper ant-input-affix-wrapper-focused"
                    >
                      <input
                        class="ant-input"
                        placeholder="Search..."
                        type="text"
                        value=""
                      />
                      <span
                        class="ant-input-suffix"
                      >
                        <span
                          aria-label="close-circle"
                          class="anticon anticon-close-circle ant-input-clear-icon-hidden ant-input-clear-icon"
                          role="button"
                          tabindex="-1"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="close-circle"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                            />
                          </svg>
                        </span>
                      </span>
                    </span>
                    <span
                      class="ant-input-group-addon"
                    >
                      <button
                        class="ant-btn ant-btn-icon-only ant-input-search-button"
                        type="button"
                      >
                        <span
                          aria-label="search"
                          class="anticon anticon-search"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="search"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                            />
                          </svg>
                        </span>
                      </button>
                    </span>
                  </span>
                </span>
              </div>
              <div>
                <div
                  class="infinite-scroll-component infinite-scroller"
                  style="height: 500px; overflow: auto;"
                >
                  <div
                    class="ant-spin-nested-loading"
                  >
                    <div>
                      <div
                        class="ant-spin ant-spin-spinning"
                      >
                        <span
                          class="ant-spin-dot ant-spin-dot-spin"
                        >
                          <i
                            class="ant-spin-dot-item"
                          />
                          <i
                            class="ant-spin-dot-item"
                          />
                          <i
                            class="ant-spin-dot-item"
                          />
                          <i
                            class="ant-spin-dot-item"
                          />
                        </span>
                      </div>
                    </div>
                    <div
                      class="ant-spin-container ant-spin-blur"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="view-container"
    >
      <div
        class="global-studio-list"
      >
        <div
          class="project-banner no-bg"
          style="margin-bottom: 20px;"
        >
          <div
            class="label"
          >
            <h1
              class="name"
            >
              <span>
                <a
                  href="/admin/org"
                  role="link"
                >
                  org
                </a>
                 | 
                <a
                  href="/admin/org/project"
                  role="link"
                >
                  project
                </a>
              </span>
            </h1>
          </div>
        </div>
        <div
          class="studio-list"
          id="studio-list"
        >
          <div
            class="studio-modal"
          >
            <button
              class="ant-btn ant-btn-primary ant-btn-block"
              type="button"
            >
              <span
                aria-label="plus-square"
                class="anticon anticon-plus-square"
                role="img"
                style="font-size: 16px; color: white;"
              >
                <svg
                  aria-hidden="true"
                  data-icon="plus-square"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M328 544h152v152c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V544h152c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H544V328c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v152H328c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8z"
                  />
                  <path
                    d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
                  />
                </svg>
              </span>
              <span>
                Create Studio
              </span>
            </button>
          </div>
          <br />
          <div
            class="infinite-search"
            data-testid="infinite-search"
          >
            <div
              class="search"
            >
              <span
                class="ant-input-group-wrapper ant-input-search"
              >
                <span
                  class="ant-input-wrapper ant-input-group"
                >
                  <span
                    class="ant-input-affix-wrapper ant-input-affix-wrapper-focused"
                  >
                    <input
                      class="ant-input"
                      placeholder="Search..."
                      type="text"
                      value=""
                    />
                    <span
                      class="ant-input-suffix"
                    >
                      <span
                        aria-label="close-circle"
                        class="anticon anticon-close-circle ant-input-clear-icon-hidden ant-input-clear-icon"
                        role="button"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="close-circle"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                          />
                        </svg>
                      </span>
                    </span>
                  </span>
                  <span
                    class="ant-input-group-addon"
                  >
                    <button
                      class="ant-btn ant-btn-icon-only ant-input-search-button"
                      type="button"
                    >
                      <span
                        aria-label="search"
                        class="anticon anticon-search"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="search"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                          />
                        </svg>
                      </span>
                    </button>
                  </span>
                </span>
              </span>
            </div>
            <div>
              <div
                class="infinite-scroll-component infinite-scroller"
                style="height: 500px; overflow: auto;"
              >
                <div
                  class="ant-spin-nested-loading"
                >
                  <div>
                    <div
                      class="ant-spin ant-spin-spinning"
                    >
                      <span
                        class="ant-spin-dot ant-spin-dot-spin"
                      >
                        <i
                          class="ant-spin-dot-item"
                        />
                        <i
                          class="ant-spin-dot-item"
                        />
                        <i
                          class="ant-spin-dot-item"
                        />
                        <i
                          class="ant-spin-dot-item"
                        />
                      </span>
                    </div>
                  </div>
                  <div
                    class="ant-spin-container ant-spin-blur"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
